### PR TITLE
Add MongoClientSettings support for MongoDbHealthCheck

### PR DESCRIPTION
**Pull request by tioma18**
Hi!
It will be good if clients are able to pass MogoClientSettings to healthcheck (for example with SSL settings).

### CODE REVIEW

**Comment by unaizorrilla** [comment]
Hi @tioma18 

I wrote some minor feedback, can you address this?

Thanks for contributting!!

File: src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs [hunk 1]
```diff
@@ -50,5 +51,48 @@ public static IHealthChecksBuilder AddMongoDb(this IHealthChecksBuilder builder,
...
+        /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+        /// <returns>The <see cref="IHealthChecksBuilder"/>.</returns></param>
+        public static IHealthChecksBuilder AddMongoDb(this IHealthChecksBuilder builder, MongoClientSettings mongoClientSettings, string mongoDatabaseName, string name = default, HealthStatus? failureStatus = default, IEnumerable<string> tags = default)
+        {
+            return builder.Add(new HealthCheckRegistration(
+                name ?? NAME,
+                sp => new MongoDbHealthCheck(mongoClientSettings, mongoDatabaseName),
+                failureStatus,
+                tags));
+        }
```
[the remark this answers is not in the document]
Comment (tioma18) [reply src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs]: > In order to mantain compatibilty can you preserve also the older AddMongoDb with connectionstring parameter and create the MongoDbSettings inside to use the same healthcheck?

Actually, it is so. Or have i misunderstood you?

File: src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs [hunk 2]
```diff
@@ -22,9 +23,10 @@ public static class MongoDbHealthCheckBuilderExtensions
...
         public static IHealthChecksBuilder AddMongoDb(this IHealthChecksBuilder builder, string mongodbConnectionString, string name = default, HealthStatus? failureStatus = default, IEnumerable<string> tags = default)
         {
+            var mongoDbHealthCheck = new MongoDbHealthCheck(mongodbConnectionString);
             return builder.Add(new HealthCheckRegistration(
                 name ?? NAME,
-                sp => new MongoDbHealthCheck(mongodbConnectionString),
+                sp => mongoDbHealthCheck,
                 failureStatus,
                 tags));
         }
```
Comment (tioma18) [inline src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs]: I wonder why you want to create MongoDbHealthCheck object every time. In it you have MongoClient which should be reused (http://mongodb.github.io/mongo-csharp-driver/2.0/reference/driver/connecting/#re-use)

File: src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (unaizorrilla) [reply src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs]: my fault, sorry!

File: src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs (hunk 2 above)
Comment (unaizorrilla) [reply src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs]: You can check ServiceBusHealthCheck in order to check how solve similar issues

Comment (tioma18) [reply src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs]: Could you provide the link? Can't find.

Comment (unaizorrilla) [reply src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs]: The idea is do something similar to 
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.AzureServiceBus/AzureServiceBusTopicHealthCheck.cs to preserve static MongoClient but enable different AddMongoDb health registrations

**Pull request merged**